api/src/modules/scenarios/cost-surface/cost-surface.facade.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { getValidSurfaceCost } from './__mocks__/surface-cost.data';
+
+let sut: CostSurfaceFacade;
+
+let costService: AdjustCostServiceFake;
+let fileConverter: ShapefileConverterFake;
+let events: CostSurfaceEventsFake;
+
+beforeEach(async () => {
+  const sandbox = await Test.createTestingModule({
+    providers: [
+      {
+        provide: AdjustCostSurface,
+        useClass: AdjustCostServiceFake,
+      },
+      {
+        provide: ResolvePuWithCost,
+        useClass: ShapefileConverterFake,
+      },
+      {
+        provide: CostSurfaceEventsPort,
+        useClass: CostSurfaceEventsFake,
+      },
+      CostSurfaceFacade,
+    ],
+  }).compile();
+
+  sut = sandbox.get(CostSurfaceFacade);
+  costService = sandbox.get(AdjustCostSurface);
+  fileConverter = sandbox.get(ResolvePuWithCost);
+  events = sandbox.get(CostSurfaceEventsPort);
+});
+
+const scenarioId = 'scenarioId';
+const request: Request = Object.freeze({
+  file: ({ fakeFile: 1 } as unknown) as Express.Multer.File,
+} as unknown) as Request;
+
+describe(`when file couldn't be converted`, () => {
+  beforeEach(async () => {
+    fileConverter.mock.mockRejectedValue(new Error('Not a shapefile.'));
+
+    // Act
+    await sut.convert(scenarioId, request);
+  });
+
+  it(`should emit relevant events`, () => {
+    expect(events.events).toMatchInlineSnapshot(`
+      Array [
+        Array [
+          "scenarioId",
+          "submitted",
+        ],
+        Array [
+          "scenarioId",
+          "shapefile-conversion-failed",
+        ],
+      ]
+    `);
+  });
+});
+
+describe(`when file can be converted`, () => {
+  beforeEach(() => {
+    fileConverter.mock.mockResolvedValue(getValidSurfaceCost());
+  });
+
+  describe(`when cost couldn't be adjusted`, () => {
+    beforeEach(async () => {
+      costService.mock.mockRejectedValue(new Error('SQL Error'));
+
+      // Act
+      await sut.convert(scenarioId, request);
+    });
+
+    it(`should emit relevant events`, () => {
+      expect(events.events).toMatchInlineSnapshot(`
+        Array [
+          Array [
+            "scenarioId",
+            "submitted",
+          ],
+          Array [
+            "scenarioId",
+            "shapefile-converted",
+          ],
+          Array [
+            "scenarioId",
+            "cost-update-failed",
+          ],
+        ]
+      `);
+    });
+  });
+
+  describe(`when cost can be adjusted`, () => {
+    beforeEach(async () => {
+      costService.mock.mockResolvedValue(undefined);
+
+      // Act
+      await sut.convert(scenarioId, request);
+    });
+
+    it(`proxies file to port`, () => {
+      expect(fileConverter.mock.mock.calls[0][0]).toEqual(request.file);
+    });
+
+    it(`proxies port output to update service`, () => {
+      expect(costService.mock.mock.calls[0][0]).toEqual(scenarioId);
+      expect(costService.mock.mock.calls[0][1]).toEqual(getValidSurfaceCost());
+    });
```

Comment:
```suggestion
    it(`updates the const with a converted cost`, () => {
      expect(costService.mock.mock.calls[0][0]).toEqual(scenarioId);
      expect(costService.mock.mock.calls[0][1]).toEqual(getValidSurfaceCost());
    });
```

the second assertion already covers that the converter is used – otherwise, it wouldn't get `getValidSurfaceCost`.

Reply:
@Dyostiq indeed it covers usage of the converter - but the assertion you suggest to remove checks if the file was passed as expected 🤔

Reply:
oh, ok

I'd modify the mock then, to return `getValidSurfaceCost` only for the file given, or keep it as is now